ctpolicy/ctpolicy.go

```diff
@@ -153,13 +177,106 @@ func (ctp *CTPolicy) GetSCTs(ctx context.Context, cert core.CertDER, expiration
 			results <- result{sct: sct}
 		}(i, g)
 	}
-	isPrecert := true
+
+	go ctp.submitPrecertInformational(cert, expiration)
+
+	var ret core.SCTDERs
+	for i := 0; i < len(ctp.groups); i++ {
+		res := <-results
+		// If any one group fails to get a SCT then we fail out immediately
+		// cancel any other in progress work as we can't continue
+		if res.err != nil {
+			// Returning triggers the defer'd context cancellation method
+			return nil, res.err
+		}
+		ret = append(ret, res.sct)
+	}
+	return ret, nil
+}
+
+// getOperatorSCTs retrieves exactly two SCTs from the total collection of
+// configured log groups, with at most one SCT coming from each group. It
+// expects that all logs run by a single operator (e.g. Google) are in the same
+// group, to guarantee that SCTs from logs in different groups do not end up
+// coming from the same operator. As such, it enforces Google's current CT
+// Policy, which requires that certs have two SCTs from logs run by different
+// operators.
+// TODO(#5938): Inline this into GetSCTs when getGoogleSCTs is removed.
+func (ctp *CTPolicy) getOperatorSCTs(ctx context.Context, cert core.CertDER, expiration time.Time) (core.SCTDERs, error) {
+	// We'll cancel this sub-context when we have the two SCTs we need, to cause
+	// any other ongoing submission attempts to quit.
+	subCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	results := make(chan result, len(ctp.operatorGroups))
+
+	// Kick off a collection of goroutines to try to submit the precert to each
+	// log operator group. Randomize the order of the groups so that we're not
+	// always trying to submit to the same two operators.
+	for i, groupIdx := range rand.Perm(len(ctp.operatorGroups)) {
+		operatorGroup := ctp.operatorGroups[groupIdx]
+		i := i
```

Comment:
That sounds like a good solution! And thanks for finding those other instances of reassignment - I was looking narrowly for `i := i`. We should probably add a coding style guideline to cover this case since it's an important source of subtle bugs.